assignment4/IslandCounter.java

```diff
@@ -0,0 +1,60 @@
+package com.katermar.assignment4;
+
+/**
+ * Created by katermar on 6/21/2017.
+ */
+public class IslandCounter {
+    private char a[][];
+
+    public IslandCounter(char[][] a) {
+        this.a = a;
+    }
+
+    /*counts number of islands in an array
+    @return number of islands
+    */
+    public int countIslands() {
+        int row = a.length;
+        int column = a[0].length;
+
+        DisjointSet disjointSet = new DisjointSet(row * column);
+
+        for (int i = 0; i < row; i++) {
+            for (int j = 0; j < column; j++) {
+                // If cell water we don't need to union it with something
+                if (a[i][j] == 'F') {
+                    continue;
+                }
+                int current = i * column + j;
+                //check 4 neighbors and union, if it is land
+                if (i + 1 < row && a[i + 1][j] == 'T') {
+                    disjointSet.union((i + 1) * column + j, current);

```

Comment:
The calculation of they key to be used in disjoint set is repeated multiple times. It would be good to extract it to a separate method.